xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -701,9 +704,15 @@
  * NOTE: This macro uses variables defined in livetable_filterProperty . It was not meant to be used alone.
  *#
 #macro (livetable_filterDateProperty)
-  ## TODO: The date filter should be smarter than this..
-  #set ($whereSql = "${whereSql} and upper(str(${safe_tableAlias}.value)) like upper(?)")
-  #set ($discard = $whereParams.add("%$filterValue%"))
+  #if ($safe_tableAlias.startsWith('doc.'))
+    #set ($dateProperty = $safe_tableAlias)
+  #else
+    #set ($dateProperty = "${safe_tableAlias}.value")
+  #end
+  #set ($timestamps = $filterValue.split('-'))
+  #set ($whereSql = "${whereSql} and $dateProperty >= ? and $dateProperty <= ?")
+  #set ($discard = $whereParams.add($datetool.toDate($numbertool.toNumber($timestamps[0]))))
+  #set ($discard = $whereParams.add($datetool.toDate($numbertool.toNumber($timestamps[1]))))
```

Comment:
Can you express "before X" or "after X" with the date range picker? The code above assumes that you always have a closed range, where both ends are specified. I think it makes sense to support also the case where one end is missing. Also, it wouldn't hurt to fall back on the string matching if the filter value doesn't have the expected range format. This is more for bulletproofing. You probably get an exception right now if the filter value is "foo".